internal/git/cherrypick.go

```diff
@@ -0,0 +1,96 @@
+package git
+
+import (
+	"emperror.dev/errors"
+	"fmt"
+	"strings"
+)
+
+type CherryPickResume string
+
+const (
+	CherryPickContinue CherryPickResume = "continue"
+	CherryPickSkip     CherryPickResume = "skip"
+	CherryPickQuit     CherryPickResume = "quit"
+	CherryPickAbort    CherryPickResume = "abort"
+)
+
+type CherryPick struct {
+	// Commits is a list of commits to apply.
+	Commits []string
+
+	// NoCommit specifies whether or not to cherry-pick without committing
+	// (equivalent to the --no-commit flag on `git cherry-pick`).
+	NoCommit bool
+
+	// FastForward specifies whether or not to fast-forward the current branch
+	// if possible (equivalent to the --ff flag on `git cherry-pick`).
+	// If true, and the parent of the commit is the current HEAD, the HEAD
+	// will be fast forwarded to the commit (instead of re-applied).
+	FastForward bool
+
+	// Resume specifies how to resume a cherry-pick operation that was
+	// interrupted by a conflict (equivalent to the --continue, --skip, --quit,
+	// and --abort flags on `git cherry-pick`).
+	// Mutually exclusive with all other options.
+	Resume CherryPickResume
+}
+
+type CherryPickResult struct {
```

Comment:
I guess initially thinking about returning this, but decided not to use this?

Reply:
Yep, good catch.